### PR TITLE
feat: add tag open expression

### PR DESCRIPTION
**Pull request by mlutze**
related to #5229

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/ast/ParsedAst.scala [hunk 1]
```diff
@@ -513,6 +513,15 @@ object ParsedAst {
       */
     case class QName(sp1: SourcePosition, name: Name.QName, sp2: SourcePosition) extends ParsedAst.Expression
 
+    /**
+      * Qualified Name Expression (reference expression).
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/ast/ParsedAst.scala:517]: comment

**Comment by magnus-madsen** [comment]
I will merge it as-is.

**Comment by magnus-madsen** [comment]
Test failures

File: main/src/ca/uwaterloo/flix/language/ast/ParsedAst.scala (hunk 1 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/ast/ParsedAst.scala:517]: Might as well fix this then @JonathanStarup

File: main/src/ca/uwaterloo/flix/language/ast/ParsedAst.scala [hunk 2]
```diff
@@ -514,7 +514,7 @@ object ParsedAst {
     case class QName(sp1: SourcePosition, name: Name.QName, sp2: SourcePosition) extends ParsedAst.Expression
 
     /**
-      * Qualified Name Expression (reference expression).
+      * An Open Qualified Name Expression (This opens the type of restrictable tags) (reference expression).
```
[the remark this answers is not in the document]
Comment (JonathanStarup) [reply main/src/ca/uwaterloo/flix/language/ast/ParsedAst.scala:517]: It still contains the name though? its not a general open expression, its a `open Tag`

[the remark this answers is not in the document]
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/ast/ParsedAst.scala:517]: okay

**Comment by JonathanStarup** [comment]
@magnus-madsen ready now

**Pull request merged**